### PR TITLE
#82 enable basic authentication 

### PR DESCRIPTION
**Pull request by NivedhaSenthil**

### CODE REVIEW

File: lib/networkHandler.js [hunk 1]
```diff
@@ -1,20 +1,48 @@
...
+        xhrEvent.emit('xhrEvent', requestPromises[p.requestId].promise);
+    }
+};
+
+const resolveXHR = p => {
+    if (requestPromises && requestPromises[p.requestId])
+        requestPromises[p.requestId].resolveFunc.call();
+};
+
+const authenticate = async (event) => {
```
Comment (zabil) [inline lib/networkHandler.js]: Let's not make this an API. Basic authentication requires base64 encoding and is sent across as a header when requesting a url. So it's better to send it as a header parameter in `goto`. Refer comment at https://github.com/getgauge/taiko/issues/82#issuecomment-408880509

Comment (NivedhaSenthil) [reply lib/networkHandler.js]: Setting authentication details as part of extra headers does not handle invalid credentials, as we can do multiple attempts to give right credentials. It has to handled separately using [AuthChallengeResponse](https://chromedevtools.github.io/devtools-protocol/tot/Network#type-AuthChallengeResponse)  which requires username and password.

Can it be sent like,

`await goto("https://postman-echo.com/basic-auth",{header: {authentication:['postman','sdg']}}); `

added with a header anyways and used separately or be like how it exists now

`await goto("https://postman-echo.com/basic-auth",authenticate('postman','sdg'));`

IMO putting it in header might be misleading as we dont use it that way and authenticate before hand. Puppeteer handles it the same way with separate [api](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pageauthenticatecredentials)

Comment (zabil) [reply lib/networkHandler.js]: Does this handle all forms of authentication apart from Basic
See https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication

Comment (NivedhaSenthil) [reply lib/networkHandler.js]: Ideally it should handle as we intercept all request with authChallenge to provide credentials but got to test it, looking for ways how to test it.

Comment (zabil) [reply lib/networkHandler.js]: Ok, because this is a challenge response. It makes more sense to have an API

Comment (NivedhaSenthil) [reply lib/networkHandler.js]: Combination of authentication will not be handled as we are getting only one pair of username and password.

**Comment by NivedhaSenthil** [comment]
Closing this PR as we will be adding auth details as extra headers.

**Pull request closed without merging**